src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/MeteredEmbeddingProvider.java

```diff
@@ -18,7 +18,7 @@
  * collect and report various metrics, such as the duration of vectorization calls and the size of
  * input texts.
  */
-public class MeteredEmbeddingProvider implements EmbeddingProvider {
+public class MeteredEmbeddingProvider extends EmbeddingProvider {
```

Comment:
Ah. So the default constructor in `EmbeddingProvider` needed for this sub-class.

Reply:
Yes, and `TestEmbeddingProvider`